### ISSUE TITLE
Cross context support

### ISSUE THREAD

**Issue by sitepark-veltrup** [eclipse/jetty.project#10617]
**Jetty version(s)**
12.0.1

**Enhancement Description**

I am trying to migrate a slightly older application from Jetty 10 to Jetty 12. This uses the Standard Tag Library (JSTL) to include parts from another context with <c:import>.

So there are two webapplications:

```
jetty-base/webapp
  + a.war
    + a.jsp
  + b.war
    + b.jsp
```

In `a/a.jsp` `b/b.jsp` should be included.

`a/a.jsp`:
```jsp
<%@ taglib uri="http://java.sun.com/jsp/jstl/core" prefix="c" %>

AAA<br>

Include b:<br>
<c:import url="/b.jsp" context="/b"/>
```

When calling `/a/a.jsp` I get the following error:
```
javax.servlet.jsp.JspTagException: Unable to get RequestDispatcher for Context: "/b" and URL: "/b.jsp". Verify values and/or enable cross context access.
```

The reason for this is probably that there is no cross context support yet.
See also
https://github.com/eclipse/jetty.project/pull/9775/files#diff-8ea47b1820cec4a9829c056dd73c32945ec5c8d24b786b913cc44cd8b96c5087R1750

Is the support planned for a future release?

**Comment by janbartel** [comment]
@gregw would you like to comment?

**Comment by gregw** [comment]
Supporting cross context dispatch is not a high priority for us to implement.    We will probably ultimately do it, but it is unlikely to be efficient nor high on our priorities. 

Tell us more about why you need cross context dispatch?  why can't your two contexts simply be a single context at root?
What kind of content is it that you dispatch to?  Does it need authentication or session access?

**Comment by sitepark-veltrup** [comment]
> Supporting cross context dispatch is not a high priority for us to implement. We will probably ultimately do it, but it is unlikely to be efficient nor high on our priorities.
> 
> Tell us more about why you need cross context dispatch? why can't your two contexts simply be a single context at root? What kind of content is it that you dispatch to? Does it need authentication or session access?

It is certainly possible to fulfill the requirements via a single context. However, it would be necessary to refactore the application. I wanted to convert the application actually only to Jetty 12.

This is used as a module system. A web application provides the basic functionality that is used in many projects. Another web application contains the customer-specific extensions.

Authentication is not needed, but session access is needed.

It would also help me if there was a way to extend `APIContext` so that I can override `getContext(String)` and I could use my implementation.

**Comment by sitepark-veltrup** [comment]
Is implemented in Jetty 12.0.8. Many thanks for that!

**Comment by gregw** [comment]
Note that not only do we now support cross context dispatch between webapps of the same EE version, we also support dispatch between version.  So an EE8 (javax) webapp can dispatch request to an EE10 (jakarta) webapp!
